src/api/app/queries/packages_for_watchlist_finder.rb

```diff
@@ -0,0 +1,10 @@
+class PackagesForWatchlistFinder
+  def initialize(relation = Package.all)
```

Comment:
```suggestion
  def initialize(relation = Package.joins(:watched_items))
```

In this watchlist finder, it makes little sense to retrieve the whole list of packages. That's why I would start with a relation that already retrieves packages inside some watchlist.

I'd do the same with all the finders.

Reply:
Sure :+1: